sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriterV2.scala

```diff
@@ -120,19 +117,18 @@ final class DataFrameWriterV2[T] private[sql](table: String, ds: Dataset[T])
   }
 
   override def create(): Unit = {
-    // create and replace could alternatively create ParsedPlan statements, like
-    // `CreateTableFromDataFrameStatement(UnresolvedRelation(tableName), ...)`, to keep the catalog
-    // resolution logic in the analyzer.
     runCommand("create") {
-      CreateTableAsSelect(
-        catalog,
-        identifier,
-        partitioning.getOrElse(Seq.empty),
+      CreateTableAsSelectStatement(
```

Comment:
This no longer passes the options as write options because the statement plans don't have write options, only table options.

This change should add `writeOptions` to the statement and update conversions to pass them through to the exec plans.

Reply:
The `Create/ReplaceTableAsSelectStatement` has an `options` field, which is used as the write options, see

https://github.com/apache/spark/blob/27d53de10faa907a53a979b66282a457767b33ba/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala#L163

https://github.com/apache/spark/blob/27d53de10faa907a53a979b66282a457767b33ba/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala#L189

https://github.com/apache/spark/blob/27d53de10faa907a53a979b66282a457767b33ba/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L296

https://github.com/apache/spark/blob/27d53de10faa907a53a979b66282a457767b33ba/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L335

Reply:
Okay, I see that we reuse options for both write options and table options.

Do we want to separate those in the v2 API? I think we only mix them in the v1 API by accident. And we set table properties using `tableProperty`. Seems reasonable to separate table options and use `tableOption` or `storageOption` to set them.

Reply:
I think about this more, and still not sure about the separation. Yes, Hive table needs this separation, but it's not general and AFAIK most databases do not separate properties and options of a table. For example, if I implement a Cassandra catalog, how shall I deal with properties and options of my `CassandraTable`?

I think another approach can be: slowly deprecate OPTIONS and only use TBLPROPERTIES. We can always set the Hive table serde properties the same as the table properties.

Anyway I think we don't need to make a decision here in this PR. I'll just add a new `writeOptions` field to the `CreateTableAsSelectStatement`.

Reply:
I think it's a good idea to deprecate table options, but that doesn't really matter here. For this PR, we should keep write options separate from table options because they are used for different things.

Reply:
yup, so my new change that adds a new `writeOption` field does separate the options of `DataFrameWriter(V2)` from the table options.